04-single-hop-6lowpan-icmp/test_spec04.py

```diff
@@ -0,0 +1,282 @@
+                         indirect=['nodes'])
+def test_task04(riot_ctrl):
+    pinger, pinged = (
+        riot_ctrl(0, APP, Shell, modules=["gnrc_pktbuf_cmd"]),
+        riot_ctrl(1, APP, Shell, modules=["gnrc_pktbuf_cmd"]),
+    )
+
+    pinged_netif, pinged_addr = lladdr(pinged.ifconfig_list())
+    pinged.ifconfig_set(pinged_netif, "channel", 26)
+    assert pinged_addr.startswith("fe80::")
+    pinger_netif, _ = lladdr(pinger.ifconfig_list())
+    pinger.ifconfig_set(pinger_netif, "channel", 26)
+
+    res = ping6(pinger, pinged_addr,
+                count=10000, interval=100, packet_size=100)
+    assert res['stats']['packet_loss'] < 10
+
+    assert pktbuf(pinged).is_empty()
+    assert pktbuf(pinger).is_empty()
+
+
+@pytest.mark.xfail(reason="Experimental task")
+@pytest.mark.local_only
+# nodes passed to riot_ctrl fixture
+@pytest.mark.parametrize('nodes',
+                         [pytest.param(['samr21-xpro', 'remote-revb'])],
+                         indirect=['nodes'])
+def test_task05(riot_ctrl):
+    pinger, pinged = (
+        riot_ctrl(0, APP, Shell, modules=["gnrc_pktbuf_cmd"]),
+        riot_ctrl(1, APP, Shell, modules=["gnrc_pktbuf_cmd"]),
+    )
+    pinged_netif, _ = lladdr(pinged.ifconfig_list())
+    pinged.ifconfig_set(pinged_netif, "channel", 17)
+    pinger_netif, _ = lladdr(pinger.ifconfig_list())
+    pinger.ifconfig_set(pinger_netif, "channel", 17)
+
+    res = ping6(pinger, "ff02::1",
+                count=1000, interval=100, packet_size=50)
+    assert res['stats']['packet_loss'] < 10
+
+    assert pktbuf(pinged).is_empty()
+    assert pktbuf(pinger).is_empty()
+
+
+@pytest.mark.xfail(reason="Experimental task")
+@pytest.mark.local_only
+# nodes passed to riot_ctrl fixture
+@pytest.mark.parametrize('nodes',
+                         [pytest.param(['samr21-xpro', 'remote-revb'])],
+                         indirect=['nodes'])
+def test_task06(riot_ctrl):
+    pinger, pinged = (
+        riot_ctrl(0, APP, Shell, modules=["gnrc_pktbuf_cmd"]),
+        riot_ctrl(1, APP, Shell, modules=["gnrc_pktbuf_cmd"]),
+    )
+    pinged_netif, pinged_addr = lladdr(pinged.ifconfig_list())
+    pinged.ifconfig_set(pinged_netif, "channel", 26)
+    assert pinged_addr.startswith("fe80::")
+    pinger_netif, _ = lladdr(pinger.ifconfig_list())
+    pinger.ifconfig_set(pinger_netif, "channel", 26)
+
+    res = ping6(pinger, pinged_addr,
+                count=1000, interval=100, packet_size=100)
+    assert res['stats']['packet_loss'] < 10
+
+    assert pktbuf(pinged).is_empty()
+    assert pktbuf(pinger).is_empty()
+
+
+@pytest.mark.xfail(reason="Experimental task")
+@pytest.mark.iotlab_creds
+# nodes passed to riot_ctrl fixture
+@pytest.mark.parametrize('nodes',
+                         [pytest.param(['samr21-xpro', 'arduino-zero'])],
+                         indirect=['nodes'])
+def test_task07(riot_ctrl):
+    pinger, pinged = (
+        riot_ctrl(0, APP, Shell, modules=["gnrc_pktbuf_cmd", "xbee"]),
```

Comment:
This is an interop test as far as I understand, only one `xbee` should be used in the test, same below for task08

```suggestion
        riot_ctrl(0, APP, Shell, modules=["gnrc_pktbuf_cmd"]),
```

Reply:
Oh right! Will remove!

Reply:
Done. (also for task 6)